backend-lib/src/main/rust/lib.rs

```diff
@@ -1489,12 +1491,37 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_proposeSh
         let network = parse_network(network_id as u32)?;
         let mut db_data = wallet_db(env, network, db_data)?;
         let account = account_id_from_jint(account)?;
+        let shielding_threshold = NonNegativeAmount::from_nonnegative_i64(shielding_threshold)
+            .map_err(|()| format_err!("Invalid shielding threshold, out of range"))?;
         let memo_bytes = env.convert_byte_array(memo).unwrap();
 
+        let transparent_receiver =
+            match utils::java_nullable_string_to_rust(env, &transparent_receiver) {
+                None => Ok(None),
+                Some(addr) => match Address::decode(&network, &addr) {
+                    Some(addr) => match addr {
+                        Address::Sapling(_) | Address::Unified(_) => Err(format_err!(
+                            "Transparent receiver is not a transparent address"
+                        )),
+                        Address::Transparent(addr) => {
+                            // TODO: Check whether the transparent receiver belongs to the account.
+                            if false {
```

Comment:
This needs implementing before the PR is merged, as it prevents the new `transparentReceiver` parameter from ever working. But it's nearly 2am and I am way too tired to finish this right now.

Reply:
Fixed.